### PR TITLE
Feature/add unused address to wallet api

### PR DESCRIPTION
**Pull request by GGAlanSmithee**
This builds on #123 and also adds `unusedAddress(): Promise<Address>` to the `WalletApi`.

There are  some things missing in this PR, that I just don't know how to implement.

I think the implementation in `selectWallet` is correct, and I've tested this. However, in `selectWalletFrom`, `selectWalletFromSeed`, and `selectWalletFromPrivateKey` I don't know what the correct implementation would be.

I am willing to attempt to implement the missing functions, but I would probably need a hint, or if you want to, you can take this PR further.

### CODE REVIEW

**Comment by alessandrokonrad** [comment]
The function `unusedAddress(): Promise<Address>` would only make sense for CIP-0030 wallets. But Lucid allows you to select a much wider range of wallets. Instead of having this function I would do something like this:
```js
const [addressHexUsed] = await api.getUsedAddresses();
const [addressHexUnused] = await api.getUnusedAddresses();
const address = addressHexUsed || addressHexUnused;
```

**Comment by GGAlanSmithee** [comment]
> Lucid allows you to select a much wider range of wallets

Ok, I am lacking some fundamental understanding regarding this, sorry.

> I would do something like this

Cool. Would this make sense in the context of Lucid? I will update the PR.

**Comment by alessandrokonrad** [comment]
> Cool. Would this make sense in the context of Lucid? I will update the PR.
> 
> That change would make #123 redundant as there should always be either a used or unused wallet if I understand it correctly. (or not redundant, but I can squash them together)

I think it's alright. I feel like no one really has figured out yet how to properly use `getUsedAddresses` and `getUnusedAddresses` from CIP-0030. I'm personally not a fan of multi address wallets anyway. It doesn't make any sense. 
I like to keep things simple and just have one address to refer to.

**Comment by GGAlanSmithee** [comment]
Yeah, it does add a lot of complexity with multiple wallets. I updated the PR to take both `usedAddress` and (if not existing) `unusedAddress` inside the `Wallet.address` function, as you recommended. I think this is a much more reasonable change, because it doesn't change the API (even though it does change the public facing functionality, so I guess you could argue it breaks backward compatability according to semver)

File: src/lucid/lucid.ts [hunk 1]
```diff
@@ -256,11 +257,24 @@ export class Lucid {
...
+
+      return unusedAddressHex;
+    }
+
     this.wallet = {
-      address: async (): Promise<Address> =>
-        C.Address.from_bytes(
-          fromHex((await api.getUsedAddresses())[0]),
-        ).to_bech32(undefined),
+      address: async (): Promise<Address | null> => {
```
Comment (alessandrokonrad) [inline src/lucid/lucid.ts:271]: We can safely assume now that we get an address back. If no address is returned then the wallet you are connected to is broken. So it should stay `Promise<Address>` and you don't need to make all these other changes.

Comment (GGAlanSmithee) [reply src/lucid/lucid.ts:271]: Right! I will fix it.

**Comment by GGAlanSmithee** [comment]
@alessandrokonrad I updated the PR according to your feedback.

**Pull request merged**